parquet-column/src/main/java/org/apache/parquet/io/vector/ObjectColumnVector.java

```diff
@@ -0,0 +1,41 @@
+/* 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.io.vector;
+
+import org.apache.parquet.column.ColumnReader;
+import org.apache.parquet.io.ColumnVector;
+
+import static java.lang.reflect.Array.newInstance;
+
+/**
+ * Vector for holding complex and application dependent types
+ */
+public class ObjectColumnVector<T> extends ColumnVector
+{
+  public final T[] values;
+
+  public ObjectColumnVector(Class<T> clazz) {
+    this.valueType = clazz;
+    values = (T[]) newInstance(clazz, MAX_VECTOR_LENGTH);
+  }
+
+  @Override
+  public void doReadFrom(ColumnReader reader, int index) {
+  }
```

Comment:
I was a little confused by this method being empty at first, but I think that this class is always used for complex (converted) objects. In that case, it's not really a `ColumnVector`. At a minimum, I think this should throw an exception because the vector is used as a `ColumnVector`, but I think the better option is to replace this with a `RecordBatch<T>` object that doesn't inherit from `ColumnVector` at all.